src/components/button/index.tsx

```diff
@@ -0,0 +1,14 @@
+import type { QwikIntrinsicElements } from "@builder.io/qwik";
+import { component$, Slot } from "@builder.io/qwik";
+
+import styles from "./button.module.css";
+
+export const Button = component$(
+  ({ class: classProp, ...props }: QwikIntrinsicElements["button"]) => {
```

Comment:
You'll see me use this pattern a lot.

In this button component, for example, we're defining our own button with its own custom styles. But if we're using the button later on, we might want to adjust some of its props, like it's class or what it does when clicked.

So by giving our custom capital-b `Button` component the same props as the lowercase-b `button` builtin, we can just forward those on.

Reply:
Ok interesting, I'll definitely be asking you some more questions about this formatting later